### PR TITLE
tinygo: fix pkg/kmodule/compression{,_tinygo}.go build tags

### PR DESCRIPTION
**Pull request by archie2x**
Add missing !tinygo line in pkg/kmodule/compression.go, without it, there are multiple definitions of the function:

    > tinygo build -tags tinygo.enable,noasm
    # github.com/u-root/u-root/pkg/kmodule
    ../../../pkg/kmodule/compression.go:20:6: 	other declaration of compressionReader
    ../../../pkg/kmodule/compression_tinygo.go:24:6: compressionReader redeclared in this block

### CODE REVIEW

**Comment by archie2x** [comment]
@leongross

File: pkg/kmodule/compression.go
Comment (hugelgupf) [inline pkg/kmodule/compression.go:4]: Can you add a newline before this, so it isn't part of the license declaration?

Comment (archie2x) [reply pkg/kmodule/compression.go:5]: done

**Review by leongross — APPROVED** [review]
lgtm

**Comment by tlaurion** [comment]
What are the current size gains of tinygoized builds vs standard?

**Comment by leongross** [comment]
Afaik there is no exhaustive size comparison since not everything is building as we would want to. I think @archie2x did some preliminary size comparisons though.

**Review by binjip978 — APPROVED** [review]

**Review by rminnich — APPROVED** [review]

**Comment by archie2x** [comment]
What does 'awaiting author' mean? I don't have write access to the repo so I can't merge...

**Comment by leongross** [comment]
This implies that the change requests are pending for the author of the PR (you). Here there was some misunderstanding and this is not correct. As I see it, after a rebase onto main this can be merged by someone with the right permissions.

**Comment by leongross** [comment]
@MDr164 @jensdrenhaus Since this is approved, can we get this merged as follow-up to https://github.com/u-root/u-root/pull/3057

**Comment by leongross** [comment]
Thanks a lot @binjip978 for the review and merge.

**Comment by archie2x** [comment]
> This implies that the change requests are pending for the author of the PR (you). Here there was some misunderstanding and this is not correct. As I see it, after a rebase onto main this can be merged by someone with the right permissions.

Thanks @leongross . Thanks for pointing this out.

**Pull request merged**
